test/addon_spec.rb

```diff
@@ -105,11 +105,29 @@
       addon2 = addons.find { |addon| addon.name == "prod2" }
 
       expect(Registration::SwMgmt).to receive(:installed_products).and_return([])
+      expect(Yast::Pkg).to receive(:ResolvableProperties).with(prod2.identifier, :product, "")
+        .and_return(["status" => :available])
       reg_not_installed_addons = Registration::Addon.registered_not_installed
 
       expect(reg_not_installed_addons.size).to eql(1)
       expect(reg_not_installed_addons.first.name).to eql(addon2.name)
     end
+
+    it "does not return not addons with not available products" do
```

Comment:
NP: there is an extra "not". Maybe _"does not return addons without available products"_